packages/kubernetes-1.20/0001-always-set-relevant-variables-for-cross-compiling.patch

```diff
@@ -0,0 +1,77 @@
+From f9efa76d8474cfe566179f5d5fa67f4e30b1db88 Mon Sep 17 00:00:00 2001
+From: Ben Cressey <bcressey@amazon.com>
+Date: Sat, 18 May 2019 16:57:12 +0000
+Subject: [PATCH] always set relevant variables for cross compiling
+
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ hack/lib/golang.sh | 52 ++++++++++++++++++++++++++--------------------
+ 1 file changed, 30 insertions(+), 22 deletions(-)
+
+diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
+index bef1d837703..204207e4fd3 100755
+--- a/hack/lib/golang.sh
++++ b/hack/lib/golang.sh
+@@ -393,29 +393,37 @@ kube::golang::set_platform_envs() {
+   export GOOS=${platform%/*}
+   export GOARCH=${platform##*/}
+ 
+-  # Do not set CC when building natively on a platform, only if cross-compiling from linux/amd64
+-  if [[ $(kube::golang::host_platform) == "linux/amd64" ]]; then
+-    # Dynamic CGO linking for other server architectures than linux/amd64 goes here
+-    # If you want to include support for more server platforms than these, add arch-specific gcc names here
+-    case "${platform}" in
+-      "linux/arm")
+-        export CGO_ENABLED=1
+-        export CC=${KUBE_LINUX_ARM_CC:-arm-linux-gnueabihf-gcc}
```

Comment:
Hmm.  In previous versions, these were hardcoded, so it made sense to unilaterally override them with the `-bottlerocket-` versions we need.  Is there any reason we'd want to allow the `$KUBE_LINUX_ARM_CC`-type overrides in our replacement code?  I don't have background on these.

Reply:
This is a bit closer to what we want, which is to always use our known cross compilers. We're still missing a way to express "always cross-compile, even if we're on linux/amd64". I expect we could come up with a patch to bridge what's left.

For now, your port of the patch looks good.